testcases/kernel/syscalls/getpeername/getpeername01.c

```diff
@@ -1,178 +1,104 @@
+
 /*
- *
- *   Copyright (c) International Business Machines  Corp., 2001
+ * Copyright (c) International Business Machines  Corp., 2001
  *   07/2001 Ported by Wayne Boyer
+ * Copyright (C) 2024 SUSE LLC Andrea Manzini <andrea.manzini@suse.com>
  *
- *   This program is free software;  you can redistribute it and/or modify
- *   it under the terms of the GNU General Public License as published by
- *   the Free Software Foundation; either version 2 of the License, or
- *   (at your option) any later version.
- *
- *   This program is distributed in the hope that it will be useful,
- *   but WITHOUT ANY WARRANTY;  without even the implied warranty of
- *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See
- *   the GNU General Public License for more details.
- *
- *   You should have received a copy of the GNU General Public License
- *   along with this program;  if not, write to the Free Software Foundation,
- *   Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
-/*
- *  Verify that getpeername() returns the proper errno for various failure cases
+/*\
+ * [Description]
+ * Verify that getpeername() returns the proper errno for various failure cases
  */
 
-#include <stdio.h>
-#include <unistd.h>
-#include <errno.h>
-#include <fcntl.h>
-#include <sys/types.h>
-#include <sys/socket.h>
-#include <sys/signal.h>
-#include <sys/ioctl.h>
-#include <netinet/in.h>
-
-#include "test.h"
-#include "safe_macros.h"
+#include "tst_test.h"
 
 static struct sockaddr_in server_addr;
 static struct sockaddr_in fsin1;
 static socklen_t sinlen;
 static socklen_t invalid_sinlen = -1;
 static int sv[2];
+static int sockfd = -1;
 
-static void setup(void);
-static void setup2(int);
-static void setup3(int);
-static void setup4(int);
-static void cleanup(void);
-static void cleanup2(int);
-static void cleanup4(int);
+static void setup_fd_file(void);
+static void setup_fd_stream(void);
+static void cleanup_fd(void);
+static void setup_pair(void);
+static void cleanup_pair(void);
 
-struct test_case_t {
-	int sockfd;
+static struct test_case {
 	struct sockaddr *sockaddr;
 	socklen_t *addrlen;
-	int expretval;
 	int experrno;
-	void (*setup) (int);
-	void (*cleanup) (int);
-	char *name;
+	void (*case_setup)(void);
+	void (*case_cleanup)(void);
 } test_cases[] = {
-	{-1, (struct sockaddr *)&fsin1, &sinlen, -1, EBADF, NULL, NULL,
-	 "EBADF"},
-	{-1, (struct sockaddr *)&fsin1, &sinlen, -1, ENOTSOCK, setup2, cleanup2,
-	 "ENOTSOCK"},
-	{-1, (struct sockaddr *)&fsin1, &sinlen, -1, ENOTCONN, setup3, cleanup2,
-	 "ENOTCONN"},
-	{-1, (struct sockaddr *)&fsin1, &invalid_sinlen, -1, EINVAL, setup4,
-	 cleanup4, "EINVAL"},
+	{(struct sockaddr *)&fsin1, &sinlen, EBADF, NULL, NULL},
+	{(struct sockaddr *)&fsin1, &sinlen, ENOTSOCK, setup_fd_file, cleanup_fd },
+	{(struct sockaddr *)&fsin1, &sinlen, ENOTCONN, setup_fd_stream, cleanup_fd },
+	{(struct sockaddr *)&fsin1, &invalid_sinlen, EINVAL, setup_pair, cleanup_pair},
```

Comment:
Struct can be initialized with struct members
```
{ .sockaddr = -1, ... }
```
Not only this is more readable but this way it helps you to avoid `NULL` and `0`, see:
https://lore.kernel.org/ltp/20240321101317.3840699-1-ruansy.fnst@fujitsu.com/

Also you can have `(struct sockaddr *)&fsin1` as a default in the test function:

```C
if (!tc->sockaddr)
    tc->sockaddr = (struct sockaddr *)&fsin1